### PR TITLE
Upgrading to OVM 2.0 (or the destruction of useless code)

### PR DESCRIPTION
**Pull request by justinjmoses**
- Removal of OVM compiler in tests and in deployment
- No more safety checking of OVM compiler or unsafe addresses
- No more `ovm-ignore.json`

### CODE REVIEW

**Review by dbeal-eth — APPROVED** [review]

File: .circleci/config.yml [hunk 1]
```diff
@@ -109,7 +100,7 @@ jobs:
       - run:
           name: Run isolated layer 2 integration tests
           command: |
-            npx hardhat test:integration:l2 --compile --deploy || true # TEMP allow pass thru till PR #1598
+            npx hardhat test:integration:l2 --compile --deploy || true
```
Comment (justinjmoses) [inline .circleci/config.yml:103]: requires rebuilding

File: .circleci/src/jobs/job-integration-tests.yml [hunk 2]
```diff
@@ -38,7 +38,7 @@ steps:
   - run:
       name: Run isolated layer 2 integration tests
       command: |
-        npx hardhat test:integration:l2 --compile --deploy || true # TEMP allow pass thru till PR #1598
+        npx hardhat test:integration:l2 --compile --deploy || true
```
Comment (justinjmoses) [inline .circleci/src/jobs/job-integration-tests.yml:41]: can you please remove the `|| true` and rebuild

File: .circleci/src/jobs/job-integration-tests.yml [hunk 3]
```diff
@@ -8,19 +8,19 @@ steps:
...
       name: Build docker containers if necessary
       command: |
         if [ ! -d ./optimism ]; then
           npx hardhat ops --fresh --build --build-ops
         fi;
-  - save_cache:
-      key: v6-optimism-build-{{ checksum "package-lock.json" }}
-      paths:
-        - ./optimism
+  # - save_cache:
```
Comment (justinjmoses) [inline .circleci/src/jobs/job-integration-tests.yml:20]: please put these back in

File: test/integration/behaviors/redeem.behavior.js [hunk 4]
```diff
@@ -34,26 +34,34 @@ function itCanRedeem({ ctx }) {
 		});
 
 		before('ensure the user has sUSD', async () => {
+			console.log('before ensure susd');
```
Comment (justinjmoses) [inline test/integration/behaviors/redeem.behavior.js:37]: could you please remove all these `console.log` commands i added :pray: ?

**Review by justinjmoses — COMMENTED** [review]
Please see comments

File: test/integration/behaviors/redeem.behavior.js (hunk 4 above)
Comment (dbeal-eth) [reply test/integration/behaviors/redeem.behavior.js:37]: lol

File: .circleci/src/jobs/job-integration-tests.yml (hunk 3 above)
Comment (dbeal-eth) [reply .circleci/src/jobs/job-integration-tests.yml]: https://github.com/Synthetixio/synthetix/pull/1598/commits/e00e59a35c3b11503a5e520954f3d54043015ee7

File: test/integration/behaviors/redeem.behavior.js (hunk 4 above)
Comment (dbeal-eth) [reply test/integration/behaviors/redeem.behavior.js]: https://github.com/Synthetixio/synthetix/pull/1598/commits/e00e59a35c3b11503a5e520954f3d54043015ee7

File: .circleci/src/jobs/job-integration-tests.yml (hunk 2 above)
Comment (dbeal-eth) [reply .circleci/src/jobs/job-integration-tests.yml]: https://github.com/Synthetixio/synthetix/pull/1598/commits/e00e59a35c3b11503a5e520954f3d54043015ee7

File: .circleci/config.yml (hunk 1 above)
Comment (dbeal-eth) [reply .circleci/config.yml]: https://github.com/Synthetixio/synthetix/pull/1598/commits/e00e59a35c3b11503a5e520954f3d54043015ee7

**Pull request merged**